File: examples/android/sync/app/src/main/java/com/mongodb/realm/examples/model/Turtle.kt
package com.mongodb.realm.examples.model

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import org.bson.types.ObjectId

open class Turtle(var name : String? = null, var age : Int = 0): RealmObject() {
    @PrimaryKey
    var _id : ObjectId = ObjectId()
    var owner : TurtleEnthusiast? = null
}

File: examples/react-native/v12/TestApp/src/models/Turtle.ts
import Realm, {BSON, ObjectSchema} from 'realm';
import {Person} from './Person';

class Turtle extends Realm.Object {
  _id!: BSON.ObjectId;
  name!: string;
  owner?: Person;
  age!: number;

  static schema: ObjectSchema = {
    name: 'Turtle',
    properties: {
      _id: 'objectId',
      name: 'string',
      owner: 'Person?',
      age: 'int',
    },
    primaryKey: '_id',
  };
}
